vochain/apputils.go

```diff
@@ -29,102 +28,118 @@ import (
 	"github.com/tendermint/tendermint/privval"
 	tmtime "github.com/tendermint/tendermint/types/time"
 	"github.com/vocdoni/storage-proofs-eth-go/ethstorageproof"
+	"github.com/vocdoni/storage-proofs-eth-go/token/mapbased"
 	"go.vocdoni.io/proto/build/go/models"
 )
 
-// CheckProof checks the validity of a census proof (depending on the origin).
-// key is the data to be proof in behalf the censusRoot.
-// In case of weighted proof, this function will return the weight as second parameter.
-func CheckProof(proof *models.Proof, censusOrigin models.CensusOrigin,
-	censusRoot, processID, key []byte) (bool, *big.Int, error) {
-	switch censusOrigin {
-	case models.CensusOrigin_OFF_CHAIN_TREE:
-		switch proof.Payload.(type) {
-		case *models.Proof_Graviton:
-			p := proof.GetGraviton()
-			if p == nil {
-				return false, nil, fmt.Errorf("graviton proof is empty")
-			}
-			valid, err := gravitontree.CheckProof(key, []byte{}, censusRoot, p.Siblings)
-			return valid, big.NewInt(1), err
-		case *models.Proof_Iden3:
-			// NOT IMPLEMENTED
-			return false, nil, fmt.Errorf("iden3 proof not implemented")
+// VerifyProofOffChainTree verifies a proof with census origin OFF_CHAIN_TREE.
+// Returns verification result and weight.
+func VerifyProofOffChainTree(process *models.Process, proof *models.Proof, censusOrigin models.CensusOrigin,
```

Comment:
Since we are refactoring this I think we should also move the Proof verification functions to a Go file (such as proof.go) instead of "apputils.go"